libobs/media-io/media-remux.c

```diff
@@ -152,6 +152,18 @@ static inline bool init_output(media_remux_job_t job, const char *out_filename)
 			// Otherwise tag 0 to let FFmpeg automatically select the appropriate tag
 			out_stream->codecpar->codec_tag = 0;
 		}
+
+		if (in_stream->codecpar->codec_type == AVMEDIA_TYPE_AUDIO) {
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(59, 24, 100)
```

Comment:
Writing mostly for my own notes.

It seems `av_get_default_channel_layout` was deprecated in avutil, not avcodec.
https://github.com/FFmpeg/FFmpeg/commit/086a8048061bf9fb4c63943f6962db48175f655c

The version of avcodec when this commit landed was 59.23.100.
https://github.com/FFmpeg/FFmpeg/blob/086a8048061bf9fb4c63943f6962db48175f655c/libavcodec/version.h#L30-L32

The version of avutil was 57.23.100.
https://github.com/FFmpeg/FFmpeg/blob/086a8048061bf9fb4c63943f6962db48175f655c/libavutil/version.h#L81-L83

I suppose this is going based off of the changelog entry where it notes that [the changes for AVCodecParameters occurred in lavc 59.24.100](https://github.com/FFmpeg/FFmpeg/blob/n6.1.1/doc/APIchanges#L352-L360) (add `ch_layout`, deprecate `channels` and `channel_layout`), and we need to account for _that_ change (which this PR is doing). The libraries are normally shipped together rather than updated independently, so this is probably fine.